### PR TITLE
Added functional test code for GA-25

### PR DESCRIPTION
**Pull request by bkpatel81**
https://tools.hmcts.net/jira/browse/GA-25

Added functional test for GA-25

- [X] Does this PR introduce a breaking change

### CODE REVIEW

**Review by sanjayparekh-hmcts — CHANGES_REQUESTED** [review]

File: src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/ProfessionalInternalUserFunctionalTest.java [hunk 1]
```diff
@@ -84,6 +88,49 @@ void testInternalUserScenario() {
...
+
+        findBySinceDatePageSizeOrAndSearchAfter(lastUpdateTimes.get(0), "3", null);
+        findBySinceDatePageSizeOrAndSearchAfter(lastUpdateTimes.get(0), "1", lastRecordIdInPage);
+        findBySinceDatePageSizeOrAndSearchAfter(lastUpdateTimes.get(0), null, lastRecordIdInPage);
+
+        findByUserIdInvalidS2SToken(invitedUserIds.get(0));
+        findByUserIdNotFound("non-existing-id");
+    }
+
+    public NewUserCreationRequest inviteUserByAnInternalOrgUser(String email) {
```
[the remark this answers is not in the document]
Comment (bkpatel81) [reply src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/ProfessionalInternalUserFunctionalTest.java:121]: Hi Sanjay, Thanks for the review. All the existing methods are in public and so I kept it same as existing standard. I hope it helps.

**Review by sanjayparekh-hmcts — APPROVED** [review]

File: src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/ProfessionalInternalUserFunctionalTest.java (hunk 1 above)
Comment (rajkatla-hmcts) [inline src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/ProfessionalInternalUserFunctionalTest.java:121]: method return type not used in caller , make it void.

File: src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/ProfessionalInternalUserFunctionalTest.java [hunk 2]
```diff
@@ -84,6 +88,49 @@ void testInternalUserScenario() {
...
+        log.info("inviteUserByAnInternalOrgUser :: STARTED");
+        NewUserCreationRequest newUserCreationRequest = professionalApiClient.createNewUserRequest(email);
+        Map<String, Object> newUserResponse = professionalApiClient.addNewUserToAnOrganisation(intActiveOrgId,
+                hmctsAdmin, newUserCreationRequest, HttpStatus.CREATED);
+        assertThat(newUserResponse).isNotNull();
+        assertThat(newUserResponse.get("userIdentifier")).isNotNull();
```
Comment (rajkatla-hmcts) [inline src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/ProfessionalInternalUserFunctionalTest.java:127]: check expected user identifier value

File: src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/AuthorizationFunctionalTest.java [hunk 3]
```diff
@@ -399,6 +399,34 @@ public void validateRetrievedUsers(Map<String, Object> searchResponse, String ex
         });
     }
 
+    public void validateRetrievedUsersDetails(Map<String, Object> searchResponse) {
+        assertThat(searchResponse.get("users")).asList().isNotEmpty();
+        assertThat(searchResponse.get("lastRecordInPage")).isNotNull();
+        assertThat(searchResponse.get("moreAvailable")).isNotNull();
+        List<HashMap> professionalUsersResponses = (List<HashMap>) searchResponse.get("users");
+
+        professionalUsersResponses.forEach(user -> {
```
Comment (rajkatla-hmcts) [inline src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/AuthorizationFunctionalTest.java:408]: check expected user count

File: src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/AuthorizationFunctionalTest.java [hunk 4]
```diff
@@ -399,6 +399,34 @@ public void validateRetrievedUsers(Map<String, Object> searchResponse, String ex
...
+            assertThat(user.get("userIdentifier")).isNotNull();
+            assertThat(user.get("lastUpdated")).isNotNull();
+
+            HashMap<String, String> orgInfo = (HashMap<String, String>) user.get("organisationInfo");
+            assertThat(orgInfo).isNotNull();
+            assertThat(orgInfo.get("organisationIdentifier")).isNotNull();
+            assertThat(orgInfo.get("status")).isNotNull();
+            assertThat(user.get("lastUpdated")).isNotNull();
+
```
Comment (rajkatla-hmcts) [inline src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/AuthorizationFunctionalTest.java:417]: check expected value for not null fields

**Review by rajkatla-hmcts — CHANGES_REQUESTED** [review]

File: src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/AuthorizationFunctionalTest.java (hunk 3 above)
Comment (bkpatel81) [reply src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/AuthorizationFunctionalTest.java:411]: Thank you for review. Its fixed.

File: src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/AuthorizationFunctionalTest.java (hunk 4 above)
Comment (bkpatel81) [reply src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/AuthorizationFunctionalTest.java:421]: Thank you for review. Raj, its common methods and generate dynamic value for multiple value so not possible to assert exact value. I have added assert for status.

File: src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/ProfessionalInternalUserFunctionalTest.java [hunk 5]
```diff
@@ -84,6 +88,49 @@ void testInternalUserScenario() {
...
+        inviteMultipleUserScenarios();
+        findUserInternalScenarios();
+    }
+
+    public void inviteMultipleUserScenarios() {
+        inviteUserByAnInternalOrgUser(generateRandomEmail());
+        inviteUserByAnInternalOrgUser(generateRandomEmail());
+        inviteUserByAnInternalOrgUser(generateRandomEmail());
+        inviteUserByAnInternalOrgUser(generateRandomEmail());
+    }
```
[the remark this answers is not in the document]
Comment (bkpatel81) [reply src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/ProfessionalInternalUserFunctionalTest.java:105]: Thank you for review. Its fixed.

File: src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/ProfessionalInternalUserFunctionalTest.java (hunk 2 above)
Comment (bkpatel81) [reply src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/ProfessionalInternalUserFunctionalTest.java:127]: Thank you for review. Its fixed. Its dynamic value generated and so cannot assert exact value at this stage.

File: src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/ProfessionalInternalUserFunctionalTest.java (hunk 1 above)
Comment (bkpatel81) [reply src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/ProfessionalInternalUserFunctionalTest.java]: fixed.

**Review by rajkatla-hmcts — APPROVED** [review]

**Pull request merged**